integrationTests/multiShard/endOfEpoch/executingEpochChange_test.go

```diff
@@ -131,3 +132,138 @@ func TestEndOfEpochChangeWithoutTransactionInMultiShardedEnvironment(t *testing.
+		numOfShards,
+		nodesPerShard,
+		numMetachainNodes,
+		integrationTests.GetConnectableAddress(advertiser),
+	)
+
+	roundsPerEpoch := int64(20)
+	for _, node := range nodes {
+		node.EndOfEpochTrigger.SetRoundsPerEpoch(roundsPerEpoch)
+	}
+
+	idxProposers := make([]int, numOfShards+1)
+	for i := 0; i < numOfShards; i++ {
+		idxProposers[i] = i * nodesPerShard
+	}
+	idxProposers[numOfShards] = numOfShards * nodesPerShard
+
+	integrationTests.DisplayAndStartNodes(nodes)
+
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Node.Stop()
+		}
+	}()
+
+	initialVal := big.NewInt(10000000)
+	sendValue := big.NewInt(5)
+	integrationTests.MintAllNodes(nodes, initialVal)
+	receiverAddress := []byte("12345678901234567890123456789012")
+
+	round := uint64(0)
+	nonce := uint64(0)
+	round = integrationTests.IncrementAndPrintRound(round)
+	nonce++
+
+	time.Sleep(time.Second)
+
+	/////////----- wait for epoch end period
+	for i := uint64(0); i <= uint64(roundsPerEpoch); i++ {
+		integrationTests.ProposeBlock(nodes, idxProposers, round, nonce)
+		integrationTests.SyncBlock(t, nodes, idxProposers, round)
+		round = integrationTests.IncrementAndPrintRound(round)
+		nonce++
+
+		for _, node := range nodes {
+			integrationTests.CreateAndSendTransaction(node, sendValue, receiverAddress, "")
+		}
+
+		time.Sleep(time.Second)
+	}
+
+	time.Sleep(time.Second)
+
+	nrRoundsToPropagateMultiShard := 10
+	for i := 0; i < nrRoundsToPropagateMultiShard; i++ {
+		integrationTests.ProposeBlock(nodes, idxProposers, round, nonce)
+		integrationTests.SyncBlock(t, nodes, idxProposers, round)
+		round = integrationTests.IncrementAndPrintRound(round)
+		nonce++
+
+		for _, node := range nodes {
+			integrationTests.CreateAndSendTransaction(node, sendValue, receiverAddress, "")
+		}
+
+		time.Sleep(time.Second)
+	}
+
+	time.Sleep(time.Second)
+
+	epoch := uint32(1)
+	/////////------ verify if current blocks are with new epoch
+	for _, node := range nodes {
+		currentShId := node.ShardCoordinator.SelfId()
+		currentHeader := node.BlockChain.GetCurrentBlockHeader()
+		assert.Equal(t, epoch, currentHeader.GetEpoch())
+
+		for _, testNode := range nodes {
+			if testNode.ShardCoordinator.SelfId() == currentShId {
+				testHeader := testNode.BlockChain.GetCurrentBlockHeader()
+				assert.Equal(t, testHeader.GetNonce(), currentHeader.GetNonce())
+			}
+		}
+	}
+
+	//////////------ verify last added shardheaders in meta are with new epoch
+	for _, node := range nodes {
+		if node.ShardCoordinator.SelfId() != sharding.MetachainShardId {
+			continue
+		}
+
+		currentMetaHdr, ok := node.BlockChain.GetCurrentBlockHeader().(*block.MetaBlock)
+		if !ok {
+			continue
+		}
+
+		shardHDrStorage := node.Storage.GetStorer(dataRetriever.BlockHeaderUnit)
+		for _, shardInfo := range currentMetaHdr.ShardInfo {
+			value, ok := node.MetaDataPool.ShardHeaders().Peek(shardInfo.HeaderHash)
+			if ok {
+				header, ok := value.(data.HeaderHandler)
+				if !ok {
```

Comment:
shouldn't you fail the test if type assertion fails?

Reply:
done.